From d3341db7e3b0de106c2cf680cc829612746aa745 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Wed, 7 Mar 2018 18:27:48 +0800
Subject: [PATCH 3442/5242] MLK-17569-2: hdp: add channel/speaker allocation
 for 4 channel

commit  78e25d870130ba5df245d90aac94cf1811016de4 from
https://source.codeaurora.org/external/imx/linux-imx.git

According to CEA-861-E section 6.6.2, add channel/speaker
allocation configuration for 4 channel.
0x0: FL, FR
0x3: FL, FR, LFE, FC
0x1F:FL, FR, LFE, FC, RL, RR, FLC, FRC

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviewed-by: Sandor Yu <sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/hdp/API_HDMI_Audio.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/mxc/hdp/API_HDMI_Audio.c b/drivers/mxc/hdp/API_HDMI_Audio.c
index b2d61ce..17a0f91 100644
--- a/drivers/mxc/hdp/API_HDMI_Audio.c
+++ b/drivers/mxc/hdp/API_HDMI_Audio.c
@@ -105,6 +105,8 @@ CDN_API_STATUS CDN_API_HDMI_AudioSetInfoFrame(state_struct *state,
 
 	if (numOfChannels == 2) {
 		packet_PB4 = 0;
+	} else if (numOfChannels == 4) {
+		packet_PB4 = 0x3;
 	} else if (numOfChannels == 8) {
 		packet_PB4 = 0x13;
 	}
-- 
1.7.9.5

